src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java

```diff
@@ -494,45 +497,86 @@ public static void close(Closeable closeable) {
     /**
      * Resolve the specified filename to a URI.
      *
-     * @param filename name os the file
+     * @param filename name of the file
      * @return resolved header file URI
      * @throws CheckstyleException on failure
      */
     public static URI getUriByFilename(String filename) throws CheckstyleException {
-        // figure out if this is a File or a URL
+        URI uri = getWebOrFilePrefixUri(filename);
+
+        if (uri == null) {
+            uri = getClasspathOrFilepathUri(filename);
```

Comment:
if someone adds this new `classpath:`, doesn't this mean this should be the first thing checked and everything else would fail anyways if they were checked?

Reply:
From private conversation with @romani, we decided that unlike the original PR for this feature, the original loading order of formats should be maintained.

Reply:
Ok. Is it possible for `classpath:` to be something else? For example, can you name a file name to start with that in linux?

Reply:
```bash
➜  TEMP cat /etc/os-release      
NAME="Manjaro Linux"
ID=manjaro
ID_LIKE=arch
BUILD_ID=rolling
PRETTY_NAME="Manjaro Linux"
ANSI_COLOR="32;1;24;144;200"
HOME_URL="https://manjaro.org/"
DOCUMENTATION_URL="https://wiki.manjaro.org/"
SUPPORT_URL="https://manjaro.org/"
BUG_REPORT_URL="https://bugs.manjaro.org/"
LOGO=manjarolinux
➜  TEMP touch "classpath:_my_new_file"
➜  TEMP ls
classpath:_my_new_file

```

Reply:
For good measure:
```
➜  TEMP touch ">my_new_file"
➜  TEMP touch "\"my_new_file"
➜  TEMP touch "|my_new_file"
➜  TEMP touch "?my_new_file"
➜  TEMP touch "*my_new_file"
➜  TEMP ls
 classpath:_my_new_file  '*my_new_file'  '?my_new_file'
'"my_new_file'           '>my_new_file'  '|my_new_file'

```

Reply:
Ok, if it can be something on an OS besides a java classpath location, then we will retain some type of backward compatibility to leave it last.